src/apps/orders/routers.py

```diff
@@ -76,14 +78,33 @@ def update_order(
     db: Session = Depends(get_db),
     request_user: User = Depends(authenticate_user),
 ) -> OrderOutputSchema:
-    return update_single_order(db, order, order_id, request_user.id)
+    order_check = get_single_order(db, order_id)
+    check_object_permission(user_id=order_check.user.id, request_user=request_user)
+    db_order = update_single_order(db, order, order_id)
+    return db_order
```

Comment:
for the future enhancements define how order will be updated, which part of order can be modified, and which can not be modified